blog/docs/articles/introduce-mqtt-source.md

```diff
@@ -0,0 +1,106 @@
+# Introducing MQTT Source with ContainerSource CRD
+
+**Author: Phuong Cao...**
+
+Knative Eventing is taking a step forward in IoT integration with the introduction of an MQTT Source. This new feature contributes to addressing a growing need in the IoT space, allowing developers to seamlessly transform MQTT messages into CloudEvents and ingest them into the Knative Eventing system.
+
+## Why MQTT Source?
+
+With the proliferation of IoT devices and edge computing, there's an increasing number of devices emitting MQTT messages to brokers. Until now, processing these messages within Knative Eventing required additional steps. The new MQTT Source bridges this gap, enabling direct ingestion and processing of MQTT messages in a cloud-native event-driven architecture.
+
+## Feature Description:
+
+Instead of developing a full-fledged controller from scratch, we've taken a simpler and more efficient approach. The MQTT Source is implemented as a container image that can be used with the existing ContainerSource CRD. This approach is similar to what we've done for the WebSocket protocol, allowing us to transform MQTT messages received into CloudEvents and send them to specified brokers.
+
+## Installations:
+
+This feature uses mqtt_paho to receive and send MQTT messages as CloudEvents. The Mosquitto Eclipse package is used to send test MQTT messages, so it can be disregarded when the feature is complete and MQTT messages are sent from another source.
+
+```shell
+go get github.com/eclipse/paho.golang/paho
+go get github.com/cloudevents/sdk-go/v2
+go get github.com/cloudevents/sdk-go/protocol/mqtt_paho/v2
+```
+
+## Using the feature:
+
+1. Set up an MQTT broker (for testing, you can use Mosquitto):
+
+```shell
+docker run -it --rm --name mosquitto -p 1883:1883 eclipse-mosquitto:2.0 mosquitto -c /mosquitto-no-auth.conf
+```
+
+2. Set up sink:
+   The MQTT Source needs a sink to receive the CloudEvent message it sends. For testing purposes, we can use an event display service to receive and display those messages. Add this event_display.yaml file.
+
+```
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+ name: event-display
+ namespace: default
+spec:
+ template:
+   spec:
+     containers:
+       - # This corresponds to
+         # https://github.com/knative/eventing/tree/main/cmd/event_display/main.go
+         image: gcr.io/knative-releases/knative.dev/eventing/cmd/event_display
+```
+
+Then keep this event_display running:
+
+```shell
+kubectl apply event-display.yaml
+```
+
+## Deploy the MQTT Source
+
+We will first need to build the Container Source using command:
+
+```shell
+ko build cmd/mqttsource/main.go
+```
```

Comment:
Here, instead of using a local dev command, can we point users to a released image?

Reply:
Just resolved the other comments! However, I'm not too sure how I can get the released image here...

Reply:
@pierDipi @Leo6Leo any ideas on how to get a nightly image for this?

Reply:
https://github.com/knative/docs/blob/main/.github/workflows/knative-docs-image.yaml

We use the GitHub action to build and publish the image to Knative's release image registry.

Reply:
@Leo6Leo we need it in eventing, not docs sadly...

Reply:
To get the image built and released we need to add `mqttsource` to the list here https://github.com/knative/eventing/blob/e4b6d6861c5b6b3696ec8d9ab82ec73e04e0a781/hack/generate-yamls.sh#L57